Starcraft2Client.py

```diff
@@ -796,6 +863,116 @@ def initialize_blank_mission_dict(location_table):
+    else:
+        display_warning(f"Couldn't find {einfo}. Please run /set_path with your SC2 install directory.", ui)
+    return False
+
+
+def check_mod_install(ui=None, debug=False) -> bool:
+    # Pull up the SC2PATH if set. If not, encourage the user to manually run check_game_install_path and troubleshoot.
+    try:
+        # Check inside the Mods folder for Archipelago.SC2Mod. If found, tell user. If not, tell user.
+        if os.path.isfile(modfile := (os.environ["SC2PATH"] / Path("Mods") / Path("Archipelago.SC2Mod"))):
+            if debug: display_success(f"Archipelago mod found at {modfile}.", ui)
+            return True
+        else:
+            display_warning(f"Archipelago mod could not be found at {modfile}. Please install the mod file there.", ui)
+    except KeyError:
+        display_warning(f"SC2PATH isn't set. Please run /check_game_install_path and troubleshoot.", ui)
+    return False
+
+
+def grab_dlls(ui=None, debug=False):
+    # Credit to Magnemania for the structure of this code.
+    # Check the lib folder of the Archipelago installation for the following files:
+    required_dll_names = {'icudt52.dll', 'icuin52.dll', 'icuuc52.dll'}
+    try:
+        dlls = set(listdir(libdir := getcwd() / Path('lib')))
+    except FileNotFoundError:
+        display_warning("Failed to check .dlls. Couldn't find the Archipelago/lib folder.", ui)
+        return False
+    missing_dlls = required_dll_names - dlls
+    if missing_dlls:
+        display_message("Missing .dlls. Attempting to grab them from your SC2 install...", ui)
+        # Without writing this to allow direct download, we can't grab the .dlls without SC2 installed.
+        try:
+            for missing_dll in missing_dlls:
+                copy2(Path(os.environ["SC2PATH"]) / "Support64" / missing_dll, libdir)
+                display_success(f"Grabbed {missing_dll}.", ui)
+        except KeyError:
+            display_warning("Failed to acquire .dlls; could not find SC2 install.", ui)
+    else:
+        if debug: display_message("No .dlls missing. Did not grab.", ui)
+
+
+def display_message(text, ui=None, tag=None):
+    # If implemented, use hex code 88CCEE for the sky blue font of paths.
+    # These colors are colorblind-accessible, found using the Tol palette at https://davidmathlogic.com/colorblind/.
+    # The grey-ish background of the client is color code 303030, for reference. It is distinguishable from these.
+    message = ""
+    if ui:
+        if tag == "warning": message += "[color=AA4499]WARNING:[/color] "
+        elif tag == "success": message += "[color=117733]SUCCESS:[/color] "
+    else:
+        if tag == "warning": message += "WARNING: "
+        elif tag == "success": message += "SUCCESS: "
+    message += text
+
+    if ui:
+        ui.log_panels['All'].on_message_markup(message)
+        ui.log_panels['Starcraft2'].on_message_markup(message)
+    else:
+        sc2_logger.info(message)
```

Comment:
This looks like you write either to GUI or to the log, which is not good because if someone requests support, the logfile should include all required info.

In other places we hook into the logger to write it to the UI. Maybe decorating/coloring the text based on the channel in a logger "hook" is the cleaner solution.